### ISSUE TITLE
Querying blueprint downs my SailsJS API  - RangeError Maximum call stack exceeded

### ISSUE THREAD

**Issue by arnthorsnaer** [balderdashy/sails#3321]
I have a SailsJS instance with blueprint actions on a few models running on top of Postgres.

I have a problem with one model in production that I don't have in development. I haven't yet tried to reproduce this with another DB or isolate this further but wanted to check here first with the community.

My location model is defined as follows 

```
module.exports = {
  attributes: {
  	name: {
  		type: 'string'
  	},
    notifications:{
      collection: 'notification',
      via: 'locations'
    },    
    topics:{
      collection: 'topic',
      via: 'location'
    }, 
    organization:{
        model:'sender'
    },
    subscriptions:{
      collection: 'subscription',
      via: 'location'
    },
    //we create the tree structure by defining superlocation attribute 
    // which in turns makes a location the sublocation of it's superlocation 
	  sublocations: {
	    collection: 'location', 
	    via: 'superlocation' 
	  },
	  superlocation: {
	    model: 'location'
	  }
  }
};
```

The queries are: /location?organization=1&populate=topics  or   /location?organization=1&populate=notifications  

I get the following error a few dozen times and then my SailsJS crashes.
(node) warning: Recursive process.nextTick detected. This will break in the next version of node. Please use setImmediate for recursive deferral. 
RangeError: Maximum call stack size exceeded

However I don't get this error when I don't include the populate=topics parameter. 

This is running on a 2 gigabyte ram VPS host (digital ocean) and I've validated that the memory usage is ~100 megabytes when SailsJS crashes.

So I am wondering. What could be recursive in regards to the "populate" command? 
I don't suppose this is related to postgres since I am also running that in development.

**Comment by crobinson42** [comment]
Do you have populate=true in your config?

**Comment by arnthorsnaer** [comment]
No, populate: false

**Comment by arnthorsnaer** [comment]
Solved this, I was running Node 0.10.25 in production and Node 0.12.7 in development. Upgrading to 0.12.7 solved this.

**Comment by mikermcneil** [comment]
@arnthorsnaer as strange as this sounds, I think this might be related to Grunt.  Not exactly sure what would have been going on re: the timing, but just wanted to add that in case it helps anyone.  (could also be complete coincidence).

If anyone sees something like this again, please open a new issue and I'll take a closer look.
